examples/textures/05-image-drawing.lisp

```diff
@@ -0,0 +1,93 @@
+                                                              :rot 0.0
+                                                              :tint +white+))
+                                 (title (make-text "PARROTS & CAT"
+                                                   300 230
+                                                   :font (asset font-asset)
+                                                   :spacing -2)))
+                            ;; Process with image manipulation functions
+                            (image-crop cat-image cat-crop)
+                            (image-flip-horizontal cat-image)
+                            (image-resize cat-image 150 200)
+                            (setf (width draw-source) (width cat-image)
+                                  (height draw-source) (height cat-image)
+                                  (width draw-dest) (* 1.5 (width cat-image))
+                                  (height draw-dest) (* 1.5 (height cat-image)))
+                            (image-draw parrots-image
+                                        cat-image
+                                        draw-source
+                                        draw-dest
+                                        +white+)
+                            (image-crop parrots-image parrot-crop)
+
+                            ;; Draw on the image with a few image draw methods
+                            (image-draw-pixel parrots-image 10 10 +raywhite+)
+                            (image-draw-circle parrots-image (make-circle 10 10 5 +raywhite+))
+                            (image-draw-rectangle parrots-image
+                                                  (make-rectangle 5 20 10 10 +raywhite+))
+
+                            ;; Draw over image using custom font
+                            (image-draw-text-ex parrots-image title)
+
+                            (let* ((half-screen-width (truncate (get-screen-width) 2))
+                                   (half-screen-height (truncate (get-screen-height) 2))
+                                   (half-texture-width (truncate (width parrots-image) 2))
+                                   (half-texture-height (truncate (height parrots-image) 2))
+                                   (x (- half-screen-width half-texture-width))
+                                   (y (- half-screen-height half-texture-height 40))
+                                   (dest (make-rectangle x y
+                                                         (width parrots-image)
+                                                         (height parrots-image)
+                                                         +white+)))
+                              (setf (dest texture) dest))
+
+                            ;; Convert to texture
+                            (load-texture-from-image parrots-image
+                                                     :texture texture)))
```

Comment:
This image manipulation stuff is a great example of code that definitely can be separated out into its own function(s).

Reply:
I'll clean this up and separate into *assets* and *scene* as soon as I get #12 working. I think that'll be soon.